brainiak/reprsimil/brsa.py

```diff
@@ -0,0 +1,1118 @@
+                F = scipy.linalg.block_diag(F, f_ele)
+            # D and F above are templates for constructing
+            # the inverse of temporal covariance matrix of noise
+
+        XTY = np.dot(design.T, Y)
+        XTDY = np.dot(np.dot(design.T, D), Y)
+        XTFY = np.dot(np.dot(design.T, F), Y)
+
+        YTY_diag = np.zeros([np.size(Y, axis=1)])
+        YTDY_diag = np.zeros([np.size(Y, axis=1)])
+        YTFY_diag = np.zeros([np.size(Y, axis=1)])
+        for i_V in range(n_V):
+            YTY_diag[i_V] = np.dot(Y[:, i_V].T, Y[:, i_V])
+            YTDY_diag[i_V] = np.dot(np.dot(Y[:, i_V].T, D), Y[:, i_V])
+            YTFY_diag[i_V] = np.dot(np.dot(Y[:, i_V].T, F), Y[:, i_V])
+
+        XTX = np.dot(design.T, design)
+        XTDX = np.dot(np.dot(design.T, D), design)
+        XTFX = np.dot(np.dot(design.T, F), design)
+        return XTY, XTDY, XTFY, YTY_diag, YTDY_diag, YTFY_diag, XTX, XTDX, XTFX
+
+    def calc_dist2_GP(self, coords=None, inten=None,
+                      GP_space=False, GP_inten=False):
+        # calculate the square of difference between each voxel's location
+        # coorinates and image intensity.
+        if GP_space:
+            assert coords is not None, 'coordinate is not provided'
+            n_V = coords.shape[1]
+            dist2 = np.zeros([n_V, n_V])
+            # square of spatial distance between every two voxels
+            for dim in range(np.size(coords, 1)):
+                coord_tile = np.tile(coords[:, dim], [n_V, 1])
+                dist2 = dist2 + (coord_tile - coord_tile.T)**2
+
+            # set the hyperparameter for the GP process:
+            if self.space_smooth_range is None:
+                space_smooth_range = np.max(dist2)**0.5 / 2.0
+            else:
+                space_smooth_range = self.space_smooth_range
+
+            if GP_inten:
+                assert inten is not None, 'intensity is not provided'
+                inten_diff2 = np.zeros([n_V, n_V])
+                # squre of difference between intensities of
+                # every two voxels
+                inten_tile = np.tile(inten, [n_V, 1])
+                inten_diff2 = (inten_tile - inten_tile.T)**2
+                # set the hyperparameter for the GP process:
+                if self.inten_smooth_range is None:
+                    inten_smooth_range = np.max(inten_diff2)**0.5 / 2.0
+                else:
+                    inten_smooth_range = self.inten_smooth_range
+                n_smooth = 2
+            else:
+                inten_diff2 = None
+                inten_smooth_range = None
+                n_smooth = 1
+        else:
+            n_smooth = 0
+            dist2 = None
+            inten_diff2 = None
+            GP_inten = False
+            space_smooth_range = None
+            inten_smooth_range = None
+        return dist2, inten_diff2, space_smooth_range, inten_smooth_range,\
+            n_smooth
+
+    def _fit_RSA_UV(self, design, Y,
```

Comment:
`X` turns into `Y` on this call (from line 264/268/272). But `design` is still `design` and doesn't turn into `X` until the intermediate variables computed in `_prepare_data`. Assuming you're going to keep the scikit learn standard in the interface and shift to your notation internally, it is probably worth (a) making the switch in a single location, and (b) calling it out explicitly.